server/internal/store/orguser.go

```diff
@@ -0,0 +1,68 @@
+package store
+
+import (
+	"errors"
+
+	v1 "github.com/llm-operator/user-manager/api/v1"
+	"gorm.io/gorm"
+)
+
+// OrganizationUser is a model for user_organization.
+type OrganizationUser struct {
+	gorm.Model
+
+	UserID         string `gorm:"uniqueIndex:user_id_org_id"`
+	OrganizationID string `gorm:"uniqueIndex:user_id_org_id"`
+
+	Role string
+}
+
+// ToProto converts the model to proto.
+func (o *OrganizationUser) ToProto() *v1.OrganizationUser {
+	return &v1.OrganizationUser{
+		UserId:         o.UserID,
+		OrganizationId: o.OrganizationID,
+		Role:           v1.Role(v1.Role_value[o.Role]),
+	}
+}
+
+// CreateOrganizationUser creates a organization user.
+func (s *S) CreateOrganizationUser(tenantID, orgID, userID, role string) (*OrganizationUser, error) {
+	org, err := s.GetOrganization(orgID)
+	if err != nil {
+		return nil, err
+	}
+	if org.TenantID != tenantID {
+		return nil, errors.New("organization not found")
```

Comment:
How about returning `gorm.ErrRecordNotFound`? Then the caller can handle the error and return `InvalidArgument` or `FailedPrecondition`.

Or we can also simply move the logic to the caller.